### PR TITLE
Ripley l

### PR DESCRIPTION
**Pull request by georgezakinih**

### CODE REVIEW

**Review by ruiheesi — COMMENTED** [review]
Hi @georgezakinih, I've reviewed the implementation and the theory behind the ripley. The function implementation looks good to me and a few comments were made mainly about documentation and potential computation with vectorized metric than loop.

I have a question on the ripley assumption on isotropy or circular clusters hinted by the radius search manner. But in data we receiving there are a lot of elongated areas. I did not have the success to find anything on it, is it something we should look into? Thanks.

File: src/spac/_ripley.py [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+    coordinates = adata.obsm[spatial_key]
+    clusters = adata.obs[cluster_key].values
+
+    mode = RipleyStat(mode)  # type: ignore[assignment]
+    if TYPE_CHECKING:
+        assert isinstance(mode, RipleyStat)
+
+    # old squidpy code
+    # N = coordinates.shape[#0]
+    N = n_observations
```
[the remark this answers is not in the document]
Comment (georgezakinih) [reply src/spac/_ripley.py]: Fixed.

File: src/spac/_ripley.py [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+
+    else:
+        for i in range(n_simulations):
+            if mode == RipleyStat.L:
+                random_i = _ppp(hull, 
+                                n_simulations=1, 
+                                n_observations=n_center+n_neighbor,
+                                seed=seed)
+               
+                # Randomly select the frist n_center cells as center cells
```
[the remark this answers is not in the document]
Comment (georgezakinih) [reply src/spac/_ripley.py:263]: random_i are the coordinates generated using the poisson point process.

File: tests/test_spatial_analysis/test_ripley.py [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+                        n_simulations=0,
+                        support=radii,
+                        area=area,
+                        phenotypes=phenotypes,
+                        copy=True)
+
+        # Get the L statistics
+        returned_l_stats = result["L_stat"]["stats"].to_numpy()
+
+        # Check that the returned statistic for the two phenotypes is "A_A"
```
[the remark this answers is not in the document]
Comment (georgezakinih) [reply tests/test_spatial_analysis/test_ripley.py]: fixed.

**Pull request merged**